addons/stock_account/models/product.py

```diff
@@ -73,11 +73,6 @@ def _is_cost_method_standard(self):
     def _set_cost_method(self):
         return self.write({'property_cost_method': self.cost_method})
```

Comment:
Has any check been made when a product changes from one category to another with different cost_method?

https://github.com/odoo/odoo/pull/18529/files#diff-e64c2b05f2fa7cb4b77eada8a18bea6bR258

Just the way it has been made at category level.